plugins/outputs/prometheus_client/prometheus_client.go

```diff
@@ -110,6 +115,21 @@ func (p *PrometheusClient) basicAuth(h http.Handler) http.Handler {
 			}
 		}
 
+		matched := false
+		remoteIPs, _, _ := net.SplitHostPort(r.RemoteAddr)
+		remoteIP := net.ParseIP(remoteIPs)
+		for _, iprange := range p.IPRange {
+			_, ipNet, _ := net.ParseCIDR(iprange)
```

Comment:
Handle the error so the next line doesn't panic if user passes bad configuration